### PR TITLE
Enable GURU_CARD_FOOTER for INTERNAL collections

### LINKED ISSUE #25: GitHub to Guru: FOOTER not working for INTERNAL collections
Add Footer optional parameter is not performing (or I am misunderstanding something about the configuration of this parameter).

My workflow

----
```
name: Create guru cards

on:
  push:

jobs:
  guru:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - uses: peckjon/github-to-guru@master
      env:
        GURU_USER_EMAIL: '${{ secrets.GURU_USER_EMAIL }}'
        GURU_USER_TOKEN: '${{ secrets.GURU_USER_TOKEN }}'
        GURU_COLLECTION_ID: '********************'
        GURU_CARD_FOOTER: 'This information is maintained by SEs in GitHub, at x repo. To request an edit on this card, message the original author listed on the corresponding document'
        GURU_CARD_YAML: ADD_TO_GURU.yaml
```
----

The card is being successfully created in Guru without the footer text included anywhere.

### PR DESCRIPTION
**Pull request by peckjon**
Fixes #25

This change makes it possible to add a footer to a card in an INTERNAL collection.
- In `copyCardData()` the line const `cardpathRegex = /__CARDPATH__/g;` is now declared at the top of the file, making it globally accessible.
- In `processStandardCollection()` the card footer is read from the `GURU_CARD_FOOTER` environment variable and, if it exists, will be appended to the card content. The text `__CARDPATH__` in the footer is replaced with the URI-encoded filename of the card.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -281,13 +282,16 @@ function processStandardCollection(auth) {
     core.setFailed("GURU_CONVERT_MARKDOWN is only supported for EXTERNAL collections: https://developer.getguru.com/docs/guru-sync-manual-api");
     return;
   } else {
+    const cardFooter = process.env.GURU_CARD_FOOTER?`\n---\n${process.env.GURU_CARD_FOOTER}\n`:'';
     let cardConfigs = yaml.parse(fs.readFileSync(process.env.GURU_CARD_YAML, 'utf8'));
     for (let cardFilename in cardConfigs) try {
+      const cardContent = fs.readFileSync(cardFilename, 'utf8');
```
Comment (maherio) [inline index.js]: looks like you're creating a const and then immediately trying to update it. should probably use `let` instead

Comment (peckjon) [reply index.js]: Gah, missed that. Thanks! Should be good to test now?

Comment (maherio) [reply index.js]: i think you updated the wrong one. `cardContent` needs to be changed to `let`

I'm still seeing the error 

I'm also not familiar with every branch in the logic so idk how relevant these are, but at a glance there are a few more consts you're trying to overwrite at some point...

* [tmpfileBase](https://github.com/peckjon/github-to-guru/pull/26/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R109) being changed [here](https://github.com/peckjon/github-to-guru/pull/26/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R111)
* [cardConfig](https://github.com/peckjon/github-to-guru/pull/26/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R126) being changed [here](https://github.com/peckjon/github-to-guru/pull/26/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R128)
* [htmlLinkMatches](https://github.com/peckjon/github-to-guru/pull/26/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R157) being changed [here](https://github.com/peckjon/github-to-guru/pull/26/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R161)
* [mdLinkMatches](https://github.com/peckjon/github-to-guru/pull/26/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R158) being changed [here](https://github.com/peckjon/github-to-guru/pull/26/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R162)

**Comment by peckjon** [comment]
https://github.com/peckjon/github-to-guru/issues/25#issuecomment-2143002950
> @maherio ❤️ THANK YOU for your diligence and patience with this. I'm obviously working too quickly and not giving this proper attention (insufficient time, yesterday's commit was on mobile).
>
> I've revised and I think mashed the const bugs (and also eliminated var entirely for let). COuld you give it another go?
>
>Also, would you be interested in being brought in as a maintainer on this project?

**Comment by maherio** [comment]
This worked!

**Pull request merged**